src/WalkingModule/src/Module.cpp

```diff
@@ -888,7 +895,8 @@ bool WalkingModule::prepareRobot(bool onTheFly)
     // get the current state of the robot
     // this is necessary because the trajectories for the joints, CoM height and neck orientation
     // depend on the current state of the robot
-    if(!m_robotControlHelper->getFeedbacksRaw(10))
+    bool getExternalRobotBase = true;
+    if(!m_robotControlHelper->getFeedbacksRaw(10, getExternalRobotBase))
```

Comment:
When it arrives to the following line we have ,
https://github.com/MiladShafiee/walking-controllers/blob/9d8f8cd3a188995a5a69bed402377cd0e7ba6094/src/RobotInterface/src/Helper.cpp#L58
 `okBaseEstimation = !m_useExternalRobotBase;` that in experiment `m_useExternalRobotBase` is `false` so  the `okBaseEstimation` will be true, therefore it does not go into the part that is related to the base data of gazebo..

Reply:
This is very error-prone and can introduce bugs in the future. An additional reason why I would prefer not to have that additional parameter in ``getFeedbacksRaw`` :thinking:

Reply:
Actually I don't have a clear idea about this because this code has been used in torque control architecture by @GiulioRomualdi but was not working in poisition-control walking and some months ago I just added this to the position control walking and I tried to tune the parameters for finding something that works! Now what are your suggestion to do? (by the way I realized that now also there is some other problems for the simulation that robots falls down and it seems there is some bug, I'm trying to investigate that)

Reply:
My comment is only that I don't like that additional parameter (``getExternalRobotBase``) passed to the method ``getFeedbacksRaw``. In fact, this was related to this other comment: https://github.com/robotology/walking-controllers/pull/52#discussion_r380131478

Reply:
> My comment is only that I don't like that additional parameter (`getExternalRobotBase`) passed to the method `getFeedbacksRaw`. In fact, this was related to this other comment: [#52 (comment)](https://github.com/robotology/walking-controllers/pull/52#discussion_r380131478)

The Boolean parameter has been removed.